packages/sitecore-jss-angular/src/components/default-empty-text-field-editing-placeholder.component.ts

```diff
@@ -1,10 +1,27 @@
-import { Component } from '@angular/core';
+import { Component, ElementRef, OnInit, Renderer2 } from '@angular/core';
 
 /**
  * Default component that will be rendered in pages when field is empty; applies for text, richtext, date and link fields.
  */
 @Component({
   selector: 'sc-default-empty-text-field-editing-placeholder',
```

Comment:
I'm trying to figure out the fix...so according to the item description

> Pages expects link field to be anchor or span element, but in Angular the technical sc-default-empty-text-field-editing-placeholder node is rendered as wrapper. Consider rework the DefaultEmptyFieldEditingComponent to use attribute selector and not generate excesive wrapper

So, you was able to use "attribute" selector for "default-image-field", however, here we have some tricks. Why we can't do the same?
In "ngOnInit" you are trying to remove base template (and wrapper?) and trying to re-create the same by adding "span"? Why we can't leave "span" in a template as it was before?
I think in general DOM manipulation is not a right way that we need to follow, it's more like a hack

Reply:
yes that's what I thought initially that making the component an attribute selector would make it work for both components but for Link, Pages editor only accepts span or anchor tag and after changing the component to attribute selector it wraps the component with div tag which again break Pages editor. Somehow we don't have the same issue for Image.

Reply:
So Pages doesn't accept "div" wrapper in all the cases except of "Image"? We need to figure out that. Can you clarify that?
If it's true, can we just render it still as "attribute selector" and remove extra "div" instead?

Reply:
So div tags are not acceptable only for link component, for rest it is acceptable.
I have refactored the implementation in a more cleaner way. Please check.